matrix/SparseVector.hpp

```diff
@@ -0,0 +1,57 @@
+#include "Vector.hpp"
+#include <cassert>
+
+template<int N> struct force_constexpr_eval {
+    static const int value = N;
+};
+
+// Vector that only store nonzero elements,
+// which indices are specified as parameter pack
+template<typename Type, int... Idxs>
```

Comment:
Right now there's actually nothing requiring that the indices are in ascending order. Maybe this is something that would be good to add? Maybe that is being overly restrictive?

Reply:
At the moment I do not see any application where it would be needed to have it in an ascending order.